### ISSUE TITLE
Lazy symbol binding failed on MacOS

### ISSUE THREAD

**Issue by IhostVlad** [WasmEdge/WasmEdge#1514]
## Description
Try launch `wasmedge` for command line executable causing following error message:
```
% wasmedge --version
dyld: lazy symbol binding failed: Symbol not found: __ZN8WasmEdge6Plugin14PluginRegisterC1EPKNS0_6Plugin16PluginDescriptorE
  Referenced from: <CUT>/.wasmedge/plugin/libwasmedgePluginWasmEdgeProcess.dylib
  Expected in: flat namespace

dyld: Symbol not found: __ZN8WasmEdge6Plugin14PluginRegisterC1EPKNS0_6Plugin16PluginDescriptorE
  Referenced from: <CUT>/.wasmedge/plugin/libwasmedgePluginWasmEdgeProcess.dylib
  Expected in: flat namespace

```

Wasmedge is installed via following instruction https://wasmedge.org/book/en/

## Environment
- Hardware Architecture: X86_64
- Operating system:  MacOS BigSur 11.6.1 
- Wasmedge 0.10.0

## Additional context 1
Mentioned symbol `__ZN8WasmEdge6Plugin14PluginRegisterC1EPKNS0_6Plugin16PluginDescriptorE` that failed to recognise is [`_WasmEdge::Plugin::PluginRegister::PluginRegister(WasmEdge::Plugin::Plugin::PluginDescriptor const*)`](https://github.com/WasmEdge/WasmEdge/blob/master/lib/plugin/plugin.cpp#L185)

However, shared library `libwasmedgePluginWasmEdgeProcess.dylib`seems to include that symbols:
```
00000000000b2160 T __ZN8WasmEdge6Plugin14PluginRegisterC1EPKNS0_6Plugin16PluginDescriptorE
00000000000b2100 T __ZN8WasmEdge6Plugin14PluginRegisterC2EPKNS0_6Plugin16PluginDescriptorE
```

## Additional context 2
Just dropping file `libwasmedgePluginWasmEdgeProcess.dylib` causing `wasmedge` ability to launch. Anyway, it does not work after default installation. Also it seems that plugins will not work in such case

**Comment by hydai** [comment]
Maybe related to this part: https://github.com/WasmEdge/WasmEdge/blob/master/plugins/wasmedge_process/CMakeLists.txt#L16-L22

We are going to investigate this. Thanks for the reporting. @ibmibmibm Could you please help with this issue?

**Comment by ibmibmibm** [comment]
Ok

**Comment by hydai** [comment]
Hi @ibmibmibm 
Just want to know if is this fixed.

**Comment by hydai** [comment]
This is fixed after 0.11.x series. Closing this.
